peregrine/run.py

```diff
@@ -98,36 +197,59 @@ def main():
                        track_results_file)
       sys.exit(1)
   else:
-    signal = load_samples(args.file,
-                          int(settings.samplingFreq*1e-3*(settings.msToProcess+22)),
-                          settings.skipNumberOfBytes,
-                          file_format=args.file_format)
-    track_results = track(signal, acq_results, settings.msToProcess)
-    try:
-      with open(track_results_file, 'wb') as f:
-        cPickle.dump(track_results, f, protocol=cPickle.HIGHEST_PROTOCOL)
-      logging.debug("Saving tracking results as '%s'" % track_results_file)
-    except IOError:
-      logging.error("Couldn't save tracking results file '%s'.",
-                    track_results_file)
+    load_samples(samples=samples,
+                 filename=args.file,
+                 file_format=args.file_format)
+
+    if ms_to_process < 0:
```

Comment:
ms_to_process set to '-1' is considered a special case, which means the the whole file needs to be processed.
There is also a comment about it in the help message, when you run peregrine with '-h'. Also see line 59 in this file.